nimble/controller/src/ble_ll_adv.c

```diff
@@ -1415,22 +1410,26 @@ ble_ll_adv_set_adv_params(uint8_t *cmd)
         props = BLE_HCI_LE_SET_EXT_ADV_PROP_LEGACY_IND;
         break;
     case BLE_HCI_ADV_TYPE_ADV_NONCONN_IND:
-        min_itvl = 0;
         props = BLE_HCI_LE_SET_EXT_ADV_PROP_LEGACY_NONCONN;
         break;
     case BLE_HCI_ADV_TYPE_ADV_SCAN_IND:
-        min_itvl = 0;
         props = BLE_HCI_LE_SET_EXT_ADV_PROP_LEGACY_SCAN;
         break;
     default:
         return BLE_ERR_INV_HCI_CMD_PARMS;
     }
 
-    /* Make sure interval minimum is valid for the advertising type */
-    if ((adv_itvl_min > adv_itvl_max) || (adv_itvl_min < min_itvl) ||
-        (adv_itvl_min > BLE_HCI_ADV_ITVL_MAX) ||
-        (adv_itvl_max > BLE_HCI_ADV_ITVL_MAX)) {
-        return BLE_ERR_INV_HCI_CMD_PARMS;
+    /* Make sure intervals values are valid
+     * (HD directed advertising ignores those parameters)
+     */
+    if (!(props & BLE_HCI_LE_SET_EXT_ADV_PROP_HD_DIRECTED)) {
+        if ((adv_itvl_min > adv_itvl_max) ||
+                (adv_itvl_min < BLE_LL_ADV_ITVL_MIN) ||
+                (adv_itvl_min > BLE_HCI_ADV_ITVL_MAX) ||
```

Comment:
would be good to use the same symbols (either HCI or LL) for both min and max (also in changes below)